libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/AdaptiveDialog.cs

```diff
@@ -406,6 +464,74 @@ protected async Task<bool> ProcessEventAsync(SequenceContext sequenceContext, Di
+        {
+            DialogEvent evt;
+            var queues = EventQueues.Read(sequenceContext);
+            var changed = queues.DequeueEvent(sequenceContext.GetState().GetValue<string>(DialogPath.LastEvent));
+            if (queues.ClearProperty.Any())
+            {
+                evt = new DialogEvent() { Name = AdaptiveEvents.ClearProperty, Value = queues.ClearProperty[0], Bubble = false };
+            }
+            else if (queues.SetProperty.Any())
+            {
+                var val = queues.SetProperty[0];
+                evt = new DialogEvent() { Name = AdaptiveEvents.SetProperty, Value = val, Bubble = false };
+
+                // TODO: For now, I'm going to dereference to a one-level array value.  There is a bug in the current code in the distinction between
+                // @ which is supposed to unwrap down to non-array and @@ which returns the whole thing. @ in the curent code works by doing [0] which
+                // is not enough.
+                var entity = val.Entity.Value;
+                if (!(entity is JArray))
+                {
+                    entity = new object[] { entity };
+                }
+
+                sequenceContext.GetState().SetValue($"{TurnPath.RECOGNIZED}.entities.{val.Entity.Name}", entity);
+            }
+            else if (queues.ChooseProperty.Any())
+            {
+                evt = new DialogEvent() { Name = AdaptiveEvents.ChooseProperty, Value = queues.ChooseProperty[0], Bubble = false };
+            }
+            else if (queues.ClarifyEntity.Any())
+            {
+                evt = new DialogEvent() { Name = AdaptiveEvents.ClarifyEntity, Value = queues.ClarifyEntity[0], Bubble = false };
+            }
+            else
+            {
+                evt = new DialogEvent() { Name = AdaptiveEvents.Ask, Bubble = false };
+            }
+
+            if (changed)
+            {
+                queues.Write(sequenceContext);
+            }
+
+            sequenceContext.GetState().SetValue(DialogPath.LastEvent, evt.Name);
+            var handled = await this.ProcessEventAsync(sequenceContext, dialogEvent: evt, preBubble: true, cancellationToken: cancellationToken).ConfigureAwait(false);
+            if (!handled)
+            {
+                // If event wasn't handled, remove it from queues and keep going if things changed
+                if (queues.DequeueEvent(evt.Name))
+                {
+                    queues.Write(sequenceContext);
+                    handled = await this.ProcessFormAsync(sequenceContext, cancellationToken);
+                }
+            }
+
+            return handled;
+        }
+
+        protected override string OnComputeId()
+        {
+            if (DebugSupport.SourceMap.TryGetValue(this, out var range))
+            {
+                return $"{GetType().Name}({Path.GetFileName(range.Path)}:{range.StartPoint.LineIndex})";
+            }
+
```

Comment:
I think we have decided this is a bad idea, as it makes it TOO responsive to changes in the code.  please remove